Sources/SwiftBSON/BSONDocument+Sequence.swift

```diff
@@ -22,6 +22,16 @@ extension BSONDocument: Sequence {
         BSONDocumentIterator(over: self.buffer)
     }
 
+    // We need to re-implement this using the default Sequence implementation since the default one from
+    // `Collection` (which `BSONDocument` also conforms to) relies on numeric indexes for iteration and is therefore
+    // very slow.
+    @inlinable
```

Comment:
I copied this declaration straight from `Sequence.swift` in the Swift standard library without really considering what it meant or did: https://github.com/apple/swift/blob/main/stdlib/public/core/Sequence.swift#L600. Having read that thread, it looks like we probably want to keep it. Note that this is the public implementation of `map` on `BSONDocument` now, so it will definitely be used across modules. Also, since users grab our source from SPM/git, I don't think we have to worry about ABI stability, since users have to recompile any of our changes anyways.

Reply:
yeah rereading I see that the stability concerns don't really apply to normal library authors. could you open a ticket about considering if there are any other places we should make inlinable down the road?

Reply:
Filed SWIFT-1069